service/qos/qos_common.cc

```diff
@@ -56,7 +60,28 @@ service_level_options service_level_options::merge_with(const service_level_opti
             }
         },
     }, ret.timeout);
+    ret.workload = std::min(ret.workload, other.workload);
```

Comment:
Does this not make unspecified override a specified workload?

Reply:
I'd say interactive > batch > unspecified.

We should have thought of a way to prevent such conflicts, but too late now. Anyway, we just need to provide something, in practice it doesn't matter (though we must make specified override unspecified).

Reply:
The easiest thing to do would be to move `unspecified` as a last field in the enum and it would naturally work as expected, but I still prefer to have `unspecified` as 0, especially if we decide to add more workload types in the future. I'll just rewrite the conflict resolution logic in v2.